nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-nar-utils/src/test/java/org/apache/nifi/nar/NativeLibFinderTest.java

```diff
@@ -0,0 +1,566 @@
+        // GIVEN
+        nativeLibDirs = new ArrayList<>();
+
+        nativeLibNameToPath = new HashMap<>();
+
+        NativeLibFinder testSubject = createTestSubject();
+        String expected = null;
+
+        // WHEN
+        String actual = testSubject.findLibrary("libName");
+
+        // THEN
+        assertEquals(expected, actual);
+        verifyNoMoreInteractions(testSubjectHelper);
+    }
+
+    @Test
+    public void testFindLibraryShouldReturnNullWhenLibNotFound() throws Exception {
+        // GIVEN
+        File nativeLibDir = mock(File.class);
+
+        nativeLibDirs = Arrays.asList(nativeLibDir);
+
+        nativeLibNameToPath = new HashMap<>();
+
+        when(testSubjectHelper.findLibrary("libName", nativeLibDir)).thenReturn(null);
+
+        NativeLibFinder testSubject = createTestSubject();
+        String expected = null;
+
+        // WHEN
+        String actual = testSubject.findLibrary("libName");
+
+        // THEN
+        assertEquals(expected, actual);
+        verify(testSubjectHelper).findLibrary("libName", nativeLibDir);
+        verifyNoMoreInteractions(testSubjectHelper);
+    }
+
+    @Test
+    public void testToDirAppliedShouldReturnParentForFile() throws Exception {
+        // GIVEN
+        Path filePath = createTempFile("mocked").toAbsolutePath();
+        File expected = filePath.getParent().toFile();
+
+        // WHEN
+        File actual = createTestSubject().toDir().apply(filePath.toFile());
+
+        // THEN
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    public void testToDirAppliedShouldReturnDirUnchanged() throws Exception {
+        // GIVEN
+        Path dirPath = createTempFile("mocked").getParent();
+        File expected = dirPath.toFile();
+
+        // WHEN
+        File actual = createTestSubject().toDir().apply(dirPath.toFile());
+
+        // THEN
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    public void testGetUsrLibDirsShouldReturnUniqueDirs() throws Exception {
+        Path dir1 = Files.createDirectory(tempDirectory.resolve("dir1"));
+        Path dir2 = Files.createDirectory(tempDirectory.resolve("dir2"));
+        Path dir3 = Files.createDirectory(tempDirectory.resolve("dir3"));
+        Path dir4 = Files.createDirectory(tempDirectory.resolve("dir4"));
+
+        Path file11 = createTempFile(dir1, "usrLib", "file11");
+        Path file12 = createTempFile(dir1, "usrLib", "file12");
+        Path file21 = createTempFile(dir2, "usrLib", "file21");
+        Path file31 = createTempFile(dir3, "usrLib", "file31");
+
+        javaLibraryPath = new HashSet<Path>() {{
+            add(file11);
+            add(file12);
+            add(file21);
+            add(file31);
+            add(dir3);
+            add(dir4);
+        }}.stream()
+                .map(Path::toFile)
+                .map(File::getAbsolutePath)
+                .collect(Collectors.joining(File.pathSeparator));
+
+        HashSet<File> expected = new HashSet<File>() {{
```

Comment:
Again, should create `Set<File> expected` and then add to it

Reply:
I can understand the previous cases but why here?
Any performance impact (be it memory or cpu related) is negligible and the code is easier on the eyes with this - otherwise not uncommon - way of initialisation.

Reply:
I do agree that this is fairly common. I personally believe that it's a bad practice because it creates a new Class unnecessarily. The cost, though, is certainly negligible. It is especially less important in a test class.